src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -3063,6 +3063,165 @@ static void F(object x)
                 Diagnostic(ErrorCode.WRN_NullReferenceArgument, "y").WithArguments("x", "void Program.F(object x)").WithLocation(11, 11));
         }
 
+        [Fact]
+        public void Directive_GloballyEnabled_GeneratedCode_DisabledByDefault()
+        {
+            var source =
+@"
+// <autogenerated />
+class Program
+{
+    static void F(object x)
+    {
+        x = null; // no warning, generated file
+    }
+}";
+            var comp = CreateCompilation(source, options: TestOptions.DebugDll.WithNullableContextOptions(NullableContextOptions.Enable));
+            comp.VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void Directive_GloballyEnabled_GeneratedCode_Enabled()
+        {
+            var source =
+@"
+// <autogenerated />
+#nullable enable
+class Program
+{
+    static void F(object x)
+    {
+        x = null; // warning
+    }
+}";
+            var comp = CreateCompilation(source, options: TestOptions.DebugDll.WithNullableContextOptions(NullableContextOptions.Enable));
+            comp.VerifyDiagnostics(
+                 // (8,13): warning CS8600: Converting null literal or possible null value to non-nullable type.
+                 //         x = null; // warning
+                 Diagnostic(ErrorCode.WRN_ConvertingNullableToNonNullable, "null").WithLocation(8, 13)
+                );
+        }
```

Comment:
>} [](start = 8, length = 1)

Consider adding similar tests (with a single `#nullable` directive) for `#nullable disable` and `#nullable restore`.